compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5987,6 +6067,62 @@ private BLangMatchTypedBindingPatternClause getSuccessPattern(BLangAccessExpress
         return successPattern;
     }
 
+    private BLangMatchTypedBindingPatternClause getSuccessPatternRecordMatch(BType type,
```

Comment:
This seems to be duplicating quite a bit of code from `getSuccessPattern`. Can we try and refactor `getSuccessPattern` instead to handle the record scenario?

Reply:
Done